packages/pwa/app/components/_app-config/index.jsx

```diff
@@ -64,12 +55,16 @@ const getLocale = (locals = {}) => {
  * You can also use the AppConfig to configure a state-management library such
  * as Redux, or Mobx, if you like.
  */
-const AppConfig = ({children, locals = {}}) => {
+const AppConfig = ({children, locals = {}, derivedConfig = {}}) => {
     const [basket, setBasket] = useState(null)
     const [customer, setCustomer] = useState(null)
 
+    // TODO: This is probably going to introduce issues as the instance of this api client
+    // is difference than the one use in the pages `getProps` methods
+    const api = new CommerceAPI({...apiConfig, ...derivedConfig})
```

Comment:
Defo something you would want to be able to correct before committing to this API change!